### PR TITLE
add icons to docs and generated tables

### PR DESCRIPTION
**Pull request by RossKen**
### Type of PR

- [ ] BUG
- [ ] FEAT
- [ ] MAINT
- [X] DOC

### Is your Pull Request linked to an existing Issue or Pull Request?

No

### Give a brief description for the solution you have provided

This PR does 2 things:

1. Adds icons for each backend to the docs
2. Adds links from comparison library tables to the function documentation

For 2, this has required some renaming of `cl` base functions to remove e.g. `LevenshteinAtThresholdsComparisonBase` -> `LevenshteinAtThresholdsBase` to more easily generate base function string from the end user functions (`levenshtein_at_thresholds` in this case). Plus I think this makes it more consistent with `cll` and `ctl` libraries. `DateDiff...` has been changed to `Datediff...` for similar reasons.

### PR Checklist

- [x] Added documentation for changes
- [ ] Added feature to example notebooks at tutorials in [splink_demos](https://github.com/moj-analytical-services/splink_demos) (if appropriate)
- [ ] Added tests (if appropriate)
- [ ] Made changes based off the latest version of Splink
- [ ] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

**Comment by RossKen** [comment]
Apologies, this is actually ready to review now.  I have now added a comparison functions table to `comparison_composition_library` just to reassure users that the functions are available across all backends

File: docs/dev_guides/changing_splink/testing.md [hunk 1]
```diff
@@ -342,19 +342,19 @@ If you need to exclude _multiple_ dialects this is also possible - just pass eac
 If you intend to write a test for a specific backend, first consider whether it is definitely specific to that backend - if not then a [backend-agnostic test](#backend-agnostic-testing) would be preferable, as then your test will be run against _many_ backends.
 If you really do need to test features peculiar to one backend, then you can write it simply as you would an ordinary `pytest` test. The only difference is that you should decorate it with `@mark_with_dialects_including` (from [tests/decorator.py](https://github.com/moj-analytical-services/splink/blob/master/tests/decorator.py)) - for example:
 
-=== "DuckDB"
+=== ":simple-duckdb: DuckDB"
```
Comment (ThomasHepworth) [inline docs/dev_guides/changing_splink/testing.md:345]: These are amazing!

File: scripts/generate_dialect_comparison_docs.py [hunk 2]
```diff
@@ -11,78 +11,123 @@
 # could always pick this up dynamically,
 # but this way we get to fix the order, and feels like not unreasonable upkeep
 dialects = ("duckdb", "spark", "athena", "sqlite", "postgres")
+dialect_table_headers = (
+    ":simple-duckdb: <br> DuckDB",
+    ":simple-apachespark: <br> Spark",
+    ":simple-amazonaws: <br> Athena",
+    ":simple-sqlite: <br> SQLite",
+    ":simple-postgresql: <br> PostgreSql",
+)
```
Comment (ThomasHepworth) [inline scripts/generate_dialect_comparison_docs.py:20]: I haven't checked this, but I'm assuming it all works w/o issue.

**Review by ThomasHepworth — APPROVED** [review]
LGTM

**Pull request merged**